src/pages/signin/SignInPage.js

```diff
@@ -112,11 +122,12 @@ function SignInPage({credentials, account}) {
         isAccountValidated: Boolean(account.validated),
         didForgetPassword: Boolean(account.forgotPassword),
         canUsePasswordlessLogins,
+        hasEmailDeliveryFailure: Boolean(account.hasEmailDeliveryFailure),
     });
 
     let welcomeHeader;
     let welcomeText;
-    if (shouldShowValidateCodeForm) {
+    if (shouldShowValidateCodeForm || shouldShowResendValidationForm) {
```

Comment:
We were somehow returning true for `shouldShowResendValidationForm` (which I think should probably be removed) but never had any logic for setting the `welcomeHeader` or `welcomeText` for that view. Now we do 😅